sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteFilesTest.java

```diff
@@ -470,15 +477,21 @@ private void testDynamicDestinationsHelper(boolean bounded) throws IOException {
     options.setTestFlag("test_value");
     Pipeline p = TestPipeline.create(options);
 
-    List<String> inputs = Lists.newArrayList("0", "1", "2", "3", "4", "5", "6", "7", "8", "9");
+    final int numInputs = 100;
+    List<String> inputs = Lists.newArrayList();
+    for (int i = 0; i < numInputs; ++i) {
+      inputs.add(Integer.toString(i));
+    }
     // Prepare timestamps for the elements.
     List<Long> timestamps = new ArrayList<>();
     for (long i = 0; i < inputs.size(); i++) {
       timestamps.add(i + 1);
     }
-
+    // If emptyShards==true make numShards larger than the number of elements per destination.
+    // This will force every destination to generate some empty shards.
```

Comment:
It's odd that the only change in the test is verifying what happens when there are not enough shards.
I thought that the behavior was also incorrect in other ways - i.e. shard numbers were generally scattered across destinations, instead of each destination having its own full set of shard numbers? Can you verify that?